app/controllers/conference_registrations_controller.rb

```diff
@@ -1,28 +1,14 @@
 class ConferenceRegistrationsController < ApplicationController
-  before_filter :authenticate_user!, except: [:new, :create]
+  before_action :authenticate_user!, unless: :allow_guests?
   load_resource :conference, find_by: :short_title
+  before_action :user_already_registered, only: :new
```

Comment:
Since this is only used in new action, I would just leave it in the action, instead of creating a new function.